sessions/react-state/box/README.md

```diff
@@ -0,0 +1,45 @@
+# React State: Box
+
+In the [`src/App.js`](./src/App.js) file, there is a button which toggles an `isActive` variable. You can check the console for its value after clicking the button.
+
+Depending on the `isActive`variable, a class called "box--active" is added to the `<div>` which should then change the `<div>`'s color.
+
+However, the color stays the same because React is not informed to rerender the App.
+
+## Task
+
+Refactor the code so that React rerenders the `App` component when `isActive` is changed.
+
+You can use the following hints as guideline:
+
+- Instead of `let`, consider the `useState` hook.
+  - Don't forget to import the hook.
+  - Its initial value should be `false`.
+- Inside of the `handleClick` function, use the `set` function of the `useState` hook to set the new value of the `isActive` state.
+
+You should now be able to toggle the boxes color with the button! 🎉
+
+## Bonus:
+
+Can you use your newly created state to toggle the text on the button itself between "Activate" and "Deactivate"?
+
```

Comment:
```suggestion

## Hacker Bonus

Check the console and pay attention to the logged value of `isActive`. You will notice, that `console.log()` doesn't log the value you set with the `set` function one line before. If you set `true` it will log `false` and vice versa. This is not a bug, but the fundamental way how state in React works.

Can you explain why this happens?

```